openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -128,21 +128,45 @@ def _get_path_of_arbitrary_backend_for_user(user):
                 return backend_path
 
     @staticmethod
-    def _is_grant_password(request):
+    def _raise_exception_if_grant_is_not_password(request):
         """
         Check if the access token provided is DOT based and has password type grant.
         """
         if is_jwt_authenticated(request):
             jwt_payload = get_decoded_jwt_from_auth(request)
             if jwt_payload['grant_type'] == dot_models.Application.GRANT_PASSWORD:
-                return True
+                return
         else:
             token_query = dot_models.AccessToken.objects.select_related('user')
             dot_token = token_query.filter(token=request.auth).first()
             if dot_token and dot_token.application.authorization_grant_type == dot_models.Application.GRANT_PASSWORD:
-                return True
+                return
 
-        return False
+        raise AuthenticationFailed({
+            'error_code': 'non_supported_token',
+            'developer_message': 'Only access tokens with grant type password are supported. '
```

Comment:
Nit: removed space at end of string.
```suggestion
            'developer_message': 'Only access tokens with grant type password are supported.'
```